### PR TITLE
loader: add config file loading

### PR DESCRIPTION
**Pull request by Abb1x**
The bootloader can now load json config files. It is still basic but it works.

### CODE REVIEW

File: sources/libs/brutal/host/efi/mem.c [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 #include <efi/srvs/bs.h>
+#include <brutal/log.h>
 
 Error host_mem_acquire(size_t size, void **out_result, MAYBE_UNUSED enum host_mem_flag flags)
 {
     auto status = efi_st()->boot_services->allocate_pool(EFI_BOOT_SERVICES_DATA, size, out_result);
 
     if (status != EFI_SUCCESS)
     {
+        panic("Failed to allocate memory: {}", status);
```
Comment (Supercip971) [inline sources/libs/brutal/host/efi/mem.c:12]: maybe print the size flags ... ?

**Review by Supercip971 — CHANGES_REQUESTED** [review]

**Review by Supercip971 — APPROVED** [review]
LGTM

File: sources/loader/main.c [hunk 2]
```diff
@@ -122,14 +124,16 @@ void loader_menu(void)
...
+
     while (key.scan_code != SCAN_ESC)
     {
         if (key.unicode_char == CHAR_CARRIAGE_RETURN)
         {
             efi_tty_clear();
-            loader_boot(&(LoaderEntry){
-                .kernel = str_cast("/kernel.elf"),
-            });
+            loader_boot(&entry, &buffer);
```
Comment (sleepy-monax) [inline sources/loader/main.c:136]: Why are you passing the buffer to loader_boot

Comment (Abb1x) [reply sources/loader/main.c:133]: to deinit it after

File: sources/loader/config.c [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#include <brutal/alloc.h>
+#include <brutal/io/buffer.h>
+#include <brutal/log.h>
+#include "brutal/alloc/global.h"
+#include "brutal/io/write.h"
+#include "brutal/text/str.h"
+#include "config.h"
```
Comment (sleepy-monax) [inline sources/loader/config.c:7]: This should be

```c
#include <brutal/alloc.h>
#include <brutal/io.h>
#include <brutal/log.h>
#include <brutal/text.h>
#include <json/json.h>
#include "loader/config.h"
```

File: sources/loader/config.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+#define JSON_MEMBER_GET(_object, _type, _name)                      \
```
Comment (sleepy-monax) [inline sources/loader/config.c:9]: This should be in brutal/json

File: sources/loader/config.h [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 #include <brutal/ds.h>
 #include <brutal/text.h>
+#include <json/json.h>
```
Comment (sleepy-monax) [inline sources/loader/config.h:5]: We don't need to include json.h here

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

**Pull request merged**